editor/src/components/canvas/canvas-utils.ts

```diff
@@ -2813,3 +2875,163 @@ function createCanvasTransientStateFromProperties(
+        throw new Error(`Unhandled drag state type ${JSON.stringify(dragState)}`)
+    }
+  }
+}
+
+export function getDragStateDrag(
+  dragState: DragState | null,
+  resizeOptions: ResizeOptions,
+): CanvasPoint | null {
+  return optionalMap((positions) => positions.drag, getDragStatePositions(dragState, resizeOptions))
+}
+
+export function getDragStateStart(
+  dragState: DragState | null,
+  resizeOptions: ResizeOptions,
+): CanvasPoint | null {
+  return optionalMap(
+    (positions) => positions.start,
+    getDragStatePositions(dragState, resizeOptions),
+  )
+}
+
+export function anyDragStarted(dragState: DragState | null): boolean {
+  if (dragState == null) {
+    return false
+  } else {
+    switch (dragState.type) {
+      case 'MOVE_DRAG_STATE':
+      case 'INSERT_DRAG_STATE':
+        return dragState.start != null
+      case 'RESIZE_DRAG_STATE':
+        return dragState.properties.some((prop) => prop.start != null)
+      default:
+        const _exhaustiveCheck: never = dragState
+        throw new Error(`Unhandled drag state type ${JSON.stringify(dragState)}`)
+    }
+  }
+}
+
+export function anyDragMovement(dragState: DragState | null): boolean {
+  if (dragState == null) {
+    return false
+  } else {
+    switch (dragState.type) {
+      case 'MOVE_DRAG_STATE':
+      case 'INSERT_DRAG_STATE':
+        return dragState.drag != null
+      case 'RESIZE_DRAG_STATE':
+        return dragState.properties.some((prop) => prop.drag != null)
+      default:
+        const _exhaustiveCheck: never = dragState
+        throw new Error(`Unhandled drag state type ${JSON.stringify(dragState)}`)
+    }
+  }
+}
+
+export function getResizeOptions(
+  flexDirection: 'horizontal' | 'vertical' | null,
+  controlDirection: 'horizontal' | 'vertical',
+): Array<LayoutTargetableProp> {
+  switch (flexDirection) {
+    case 'horizontal':
+      switch (controlDirection) {
+        case 'horizontal':
+          return ['Height', 'minHeight', 'maxHeight']
+        case 'vertical':
+          return ['flexBasis', 'flexGrow', 'flexShrink', 'minWidth', 'maxWidth']
+        default:
+          const _exhaustiveCheck: never = controlDirection
+          throw new Error(`Unhandled control direction ${JSON.stringify(controlDirection)}`)
+      }
+    case 'vertical':
+      switch (controlDirection) {
+        case 'horizontal':
+          return ['flexBasis', 'flexGrow', 'flexShrink', 'minHeight', 'maxHeight']
+        case 'vertical':
+          return ['Width', 'minWidth', 'maxWidth']
+        default:
+          const _exhaustiveCheck: never = controlDirection
+          throw new Error(`Unhandled control direction ${JSON.stringify(controlDirection)}`)
+      }
+    case null:
+      switch (controlDirection) {
+        case 'horizontal':
+          return ['Height', 'marginTop', 'marginBottom', 'minHeight', 'maxHeight']
+        case 'vertical':
+          return ['Width', 'marginLeft', 'marginRight', 'minWidth', 'maxWidth']
```

Comment:
Looks like these should also include `left`, `right`, `top` and `bottom`

Reply:
Also why are the `controlDirection`s inverted?

Reply:
> Looks like these should also include `left`, `right`, `top` and `bottom`

Why? Those weren't in either the experiment or in #1654.

Reply:
> Also why are the `controlDirection`s inverted?

That's how they are seemingly: https://github.com/concrete-utopia/utopia/blob/feature/multi-option-resize/editor/src/components/canvas/controls/size-box.tsx#L617-L618

Reply:
> > Looks like these should also include `left`, `right`, `top` and `bottom`
> 
> Why? Those weren't in either the experiment or in #1654.

Because I can no longer drag to set those values if they've been set and didn't realise that was the case with this menu. Now if I have `left` and `right` set, dragging this control will add `width`, whereas dragging the corner controls will update the correct props.